### PR TITLE
Enhance Playlist player controls

### PR DESCRIPTION
**Pull request by sangwoo108**
Resolves https://github.com/brave/brave-browser/issues/34216

- The next/prev button : If it's shorter than a 10 minute file, start from the beginning, otherwise start form the last played position
- Don't change items order even when it's shuffled
  - We were showing shuffled order to users but it's unnecessary.
- The loop button
  1. play list in a loop, 
  2. play one song on repeat, 
  3. reset/original state

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/playlist/browser/resources/components/playerControls.tsx [hunk 1]
```diff
@@ -149,31 +162,49 @@ export default function PlayerControls ({ videoElement, className }: Props) {
...
+          iconName={
+            !loopMode
+              ? 'loop-all'
+              : loopMode === 'single-item'
+              ? 'loop-1-toggle-on'
+              : 'loop-all-toggle-on'
+          }
+          size='large'
+          visibility='normal'
+          title={'loop'}
```
Comment (fallaciousreasoning) [inline components/playlist/browser/resources/components/playerControls.tsx:204]: Does this need translating?

File: components/playlist/browser/resources/reducers/player_reducer.ts [hunk 2]
```diff
@@ -120,9 +121,19 @@ const playerReducer: Reducer<PlayerState | undefined> = (
...
         if (currentIndex !== items.length - 1) {
-          const currentItem = items[currentIndex + 1]
-          currentItem.lastPlayedPosition = 0 // Don't resume at the last position unless user explicitly selects an item
+          currentItem = items[currentIndex + 1]
+        } else {
+          currentItem = items[0]
+        }
+
+        if (currentItem) {
+          console.log(+currentItem.duration)
```
Comment (fallaciousreasoning) [inline components/playlist/browser/resources/reducers/player_reducer.ts:132]: nit: can we remove this log?

File: components/playlist/browser/resources/reducers/player_reducer.ts [hunk 3]
```diff
@@ -120,9 +121,19 @@ const playerReducer: Reducer<PlayerState | undefined> = (
           throw new Error("Couldn't find the index of the current item ")
         }
 
+        let currentItem = null
```
Comment (fallaciousreasoning) [inline components/playlist/browser/resources/reducers/player_reducer.ts:124]: Optional:

```suggestion
const currentItem = items[(currentIndex+1)%items.length]
```

File: components/playlist/browser/resources/reducers/player_reducer.ts [hunk 4]
```diff
@@ -136,9 +147,18 @@ const playerReducer: Reducer<PlayerState | undefined> = (
...
```
Comment (fallaciousreasoning) [inline components/playlist/browser/resources/reducers/player_reducer.ts:150]: Optional:
```suggestion
        const currentItem = items.at(currentIndex - 1)
```

**Review by fallaciousreasoning — APPROVED** [review]
LGTM

File: components/playlist/browser/resources/reducers/player_reducer.ts (hunk 4 above)
Comment (sangwoo108) [reply components/playlist/browser/resources/reducers/player_reducer.ts:150]: awesome, I completely forgot about the negative index.

File: components/playlist/browser/resources/components/playerControls.tsx (hunk 1 above)
Comment (sangwoo108) [reply components/playlist/browser/resources/components/playerControls.tsx:204]: Oh Yeah, right. Thanks!

**Pull request merged**